app/cookies/ServersideAbTestCookie.scala

```diff
@@ -0,0 +1,44 @@
+package cookies
+
+import admin.ServersideAbTest
+import admin.ServersideAbTest.Participation
+import config.Configuration.GuardianDomain
+import play.api.mvc.{Cookie, RequestHeader, Result}
+
+object ServersideAbTestCookie {
+  val name = "gu.serverside.ab.test"
+
+  def create(domain: GuardianDomain, participation: ServersideAbTest.Participation): Cookie = Cookie(
+    name = name,
+    value = participation.toString,
+    secure = true,
+    httpOnly = false,
+    domain = Some(s"support${domain.value}")
+  )
+
+  def exists(implicit request: RequestHeader): Boolean = request.cookies.exists(_.name == name)
+  def get(implicit request: RequestHeader): Option[Cookie] = request.cookies.get(name)
+
+  def addTo(result: Result)(implicit domain: GuardianDomain, participation: Participation): Result =
+    result.withCookies(create(domain, participation))
+}
+
+trait ServersideAbTestCookie {
+
+  implicit class ResultSyntax(result: Result)(
```

Comment:
I also struggle with implicit classes - I do use them reluctantly a bit to move boilerplate to the end of for comprehensions (e.g. we have to lift various things into a EitherT[Future[Either[.]]] as part of a for comprehension) but on the whole, a normal function suffices for almost everything.

Reply:
Yeah I'm inclined to agree. Though I'm following the pattern of `.withSettingsSurrogateKey` here and I feel like they should at least be consistent - and I'm not prepared to switch over all the `withSettingsSurrogateKey` usages in this PR. I will come back to this in a future PR